cmd/snap/cmd_routine_file_access.go

```diff
@@ -0,0 +1,215 @@
+despite being readable under a different path).  It also does not
+check if file system permissions would render a file unreadable.
+`)
+
+func init() {
+	addRoutineCommand("file-access", shortRoutineFileAccessHelp, longRoutineFileAccessHelp, func() flags.Commander {
+		return &cmdRoutineFileAccess{}
+	}, nil, []argDesc{
+		{
+			// TRANSLATORS: This needs to begin with < and end with >
+			name: i18n.G("<snap>"),
+			// TRANSLATORS: This should not start with a lowercase letter.
+			desc: i18n.G("Snap name"),
+		},
+		{
+			// TRANSLATORS: This needs to begin with < and end with >
+			name: i18n.G("<path>"),
+			// TRANSLATORS: This should not start with a lowercase letter.
+			desc: i18n.G("File path"),
+		},
+	})
+}
+
+func (x *cmdRoutineFileAccess) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	snapName := string(x.FileAccessOptions.Snap)
+	path := string(x.FileAccessOptions.Path)
+
+	snap, _, err := x.client.Snap(snapName)
+	if err != nil {
+		return fmt.Errorf("cannot retrieve info for snap %q: %v", snapName, err)
+	}
+
+	// Check whether the snap has home or removable-media plugs connected
+	connections, err := x.client.Connections(&client.ConnectionOptions{
+		Snap: snap.Name,
+	})
+	if err != nil {
+		return fmt.Errorf("cannot get connections for snap %q: %v", snap.Name, err)
+	}
+	var hasHome, hasRemovableMedia bool
+	for _, conn := range connections.Established {
+		if conn.Plug.Snap != snap.Name {
+			continue
+		}
+		switch conn.Interface {
+		case "home":
+			hasHome = true
+		case "removable-media":
+			hasRemovableMedia = true
+		}
+	}
+
+	access, err := x.checkAccess(snap, hasHome, hasRemovableMedia, path)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintln(Stdout, access)
+	return nil
+}
+
+type FileAccess string
+
+const (
+	FileAccessHidden    FileAccess = "hidden"
+	FileAccessReadOnly  FileAccess = "read-only"
+	FileAccessReadWrite FileAccess = "read-write"
+)
+
+func splitPath(path string) ([]string, error) {
+	// Abs also cleans the path, removing any ".." components
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return nil, err
+	}
+	// Ignore the component before the first slash
+	return strings.Split(path, string(os.PathSeparator))[1:], nil
+}
+
+func pathHasPrefix(path, prefix []string) bool {
+	if len(path) < len(prefix) {
+		return false
+	}
+	for i := range prefix {
+		if path[i] != prefix[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func (x *cmdRoutineFileAccess) checkAccess(snap *client.Snap, hasHome, hasRemovableMedia bool, path string) (FileAccess, error) {
+	// Classic confinement snaps run in the host system namespace,
+	// so can see everything.
+	if snap.Confinement == client.ClassicConfinement {
+		return FileAccessReadWrite, nil
+	}
+
+	pathParts, err := splitPath(path)
+	if err != nil {
+		return "", err
+	}
+
+	// Snaps have access to $SNAP_DATA and $SNAP_COMMON
+	if pathHasPrefix(pathParts, []string{"var", "snap", snap.Name}) {
```

Comment:
You probably need to make this snap instance aware, unless snap.Name includes both `name_key`

Reply:
This is `client.Snap.Name`, which is derived from `snap.Info.InstanceName()`.  So wouldn't this already be parallel instance aware?